### ISSUE TITLE
Sometimes it is not possible to move past a stop point and trigger the boss.

### ISSUE THREAD

**Issue by odecay** [fishfolks/punchy#243]
### Description

May be system ordering dependent or involve the stop point system.

https://user-images.githubusercontent.com/1379590/187258683-c44bd4a0-1d0f-4ed2-9e01-6e0c737cc069.mp4

### To Reproduce

Clear all enemies, then walk to the right. Only reproducible sometimes, may be per run system ordering dependent.

### Expected Behavior

Expect to be able to trigger boss by walking past the stop point.

**Comment by 64kramsystem** [comment]
Hmmm :thinking: I'll take a look!

**Comment by 64kramsystem** [comment]
What happens in cases like this if you move the player that it's behind? In the (first) iteration of the scrolling, as  I've implemented it, not allowing the players to move in the whole area of the screen was intentional, in order to keep the logic simple. If this is the problem you refer to, then we should improve the scrolling routine sooner rather than later :grimacing:.

**Comment by 64kramsystem** [comment]
> edit: yeap seems like you are right, in which case this is not really a bug. I think if p2 gets knocked backwards a bit from starting position by the enemies it prevents p1 from moving past the trigger point for boss. I notice now in the video I posted that sharky is a bit behind its normal starting position, compared by visual reference to the rock above.

Yeah, the player that is behind inevitably blocks the scrolling. When I implemented this, there was no boss, so it was acceptable to keep it simple. The difference is that, now that we have boss fights, the players should be allowed to move over the whole screen (currently, there is a limit). It should be fairly easy to add this extra constraint, although I haven't given much thought about how to keep it as simple as possible (that is, avoid a jungle of if/else :laughing:). A good issue for 0.5.0 :smile:

**Comment by 64kramsystem** [comment]
I think we should change the wording of this, to make it clear that it's an intended upgrade of a current functionality.

**Comment by odecay** [comment]
> I think we should change the wording of this, to make it clear that it's an intended upgrade of a current functionality, so that it's also more obvious for anyone who wanted to pick it up.

I agree. If you want to edit the issue feel free.

**Comment by 64kramsystem** [comment]
Ok. I've updated the description, and let the video. I think the logic described above should be what required, in the simplest possible terms (if you think there are other cases, please add!).
